### PR TITLE
SMTP per User Configuration

### PR DESCRIPTION
**Pull request by mocodertz**

### CODE REVIEW

File: modules/Emails/mail.php [hunk 1]
```diff
@@ -307,29 +307,48 @@ function setMailerProperties($mail, $subject, $contents, $from_email, $from_name
   * $mail -- reference of the mailobject
   */
 function setMailServerProperties($mail) {
-	global $adb,$default_charset;
+	global $adb,$default_charset, $current_user;
 	$adb->println('> setMailServerProperties');
-
+	$user_mail_config = $adb->pquery('select * from vtiger_mail_accounts where user_id=?', array($current_user->id));
```
Comment (joebordes) [inline modules/Emails/mail.php]: you are not checking that the outgoing email server has been configured. Imagine a user who configures incoming email server but does not configure outgoing email server. In that case, there will be one row returned but not valid outgoing email server settings.   Am I correct?

File: include/integrations/smtp/smtp.php [hunk 2]
```diff
@@ -0,0 +1,402 @@
+<?php
+/*************************************************************************************************
+ * Copyright 2017 JPL TSolucio, S.L. -- This file is a part of TSOLUCIO coreBOS customizations.
```
Comment (joebordes) [inline include/integrations/smtp/smtp.php]: 2019

File: include/integrations/smtp/settings.php [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+ * License. Roughly speaking, non-commercial users may share and modify this code, but must give credit
+ * and share improvements. However, for proper details please read the full License, available at
+ * http://vizsage.com/license/Vizsage-License-BY-NC-SA.html and the handy reference for understanding
+ * the full license at http://vizsage.com/license/Vizsage-Deed-BY-NC-SA.html. Unless required by
+ * applicable law or agreed to in writing, any software distributed under the License is distributed
+ * on an  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the
+ * License terms of Creative Commons Attribution-NonCommercial-ShareAlike 3.0 (the License).
+ *************************************************************************************************
+ *  Module    : SendGrid Integration Settings
```
Comment (joebordes) [inline include/integrations/smtp/settings.php]: SMTP

File: build/changeSets/2019/addColumnsForOutGoingServerConfigurationValues.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<?php
+/*************************************************************************************************
+ * Copyright 2017 JPL TSolucio, S.L. -- This file is a part of TSOLUCIO coreBOS Customizations.
```
Comment (joebordes) [inline build/changeSets/2019/addColumnsForOutGoingServerConfigurationValues.php]: 2019

**Comment by joebordes** [comment]
wow, looks more complex than I imagined
I added some comments and the CI found some formatting issues on the JS file, fix those and I will test it

thanks

File: modules/Emails/mail.php (hunk 1 above)
Comment (mocodertz) [reply modules/Emails/mail.php]: absolutely correct

**Comment by joebordes** [comment]
thanks!

**Pull request merged**